manager/controlapi/service.go

```diff
@@ -215,17 +215,20 @@ func (s *Server) checkPortConflicts(spec *api.ServiceSpec) error {
 	}
 
 	for _, service := range services {
-		if service.Spec.Endpoint != nil {
-			for _, pc := range service.Spec.Endpoint.Ports {
-				if reqPorts[pcToString(pc)] {
-					return grpc.Errorf(codes.InvalidArgument, "port '%d' is already in use by service '%s' (%s)", pc.PublishedPort, service.Spec.Annotations.Name, service.ID)
+		// If service ID is the same then this is an update
+		if serviceID != service.ID {
```

Comment:
To avoid further netsting, can you change this to `if serviceID == service.ID { continue }`